src/services/jsDoc.ts

```diff
@@ -269,34 +269,35 @@ namespace ts.JsDoc {
         if (!commentOwnerInfo) {
             return undefined;
         }
-        const { commentOwner, parameters } = commentOwnerInfo;
+
+        const { commentOwner, parameters, hasReturn } = commentOwnerInfo;
         if (commentOwner.getStart(sourceFile) < position) {
             return undefined;
         }
 
-        if (!parameters || parameters.length === 0) {
-            // if there are no parameters, just complete to a single line JSDoc comment
-            const singleLineResult = "/** */";
-            return { newText: singleLineResult, caretOffset: 3 };
-        }
-
         const indentationStr = getIndentationStringAtPosition(sourceFile, position);
+        const isJavaScriptFile = hasJSFileExtension(sourceFile.fileName);
+        const tags =
+            (parameters ? parameterDocComments(parameters || [], isJavaScriptFile, indentationStr, newLine) : "") +
+            (hasReturn ? returnsDocComment(isJavaScriptFile, indentationStr, newLine) : "");
 
         // A doc comment consists of the following
         // * The opening comment line
         // * the first line (without a param) for the object's untagged info (this is also where the caret ends up)
         // * the '@param'-tagged lines
+        // * the '@returns'-tag
         // * TODO: other tags.
         // * the closing comment line
         // * if the caret was directly in front of the object, then we add an extra line and indentation.
-        const preamble = "/**" + newLine + indentationStr + " * ";
-        const result =
-            preamble + newLine +
-            parameterDocComments(parameters, hasJSFileExtension(sourceFile.fileName), indentationStr, newLine) +
-            indentationStr + " */" +
-            (tokenStart === position ? newLine + indentationStr : "");
-
-        return { newText: result, caretOffset: preamble.length };
+        const openComment = "/**";
+        const closeComment = " */";
+        if (tags) {
+            const preamble = openComment + newLine + indentationStr + " * ";
+            const endLine = tokenStart === position ? newLine + indentationStr : "";
+            const result = preamble + newLine + tags + indentationStr + closeComment + endLine;
+            return { newText: result, caretOffset: preamble.length };
+        }
+        return { newText: openComment + closeComment, caretOffset: 3 };

```

Comment:
The `caretOffset` was previously calculated using `newLine` which is system dependent. Seems like changing it would cause problems, no?

Reply:
The following code that was used before

```ts
const singleLineResult = "/** */";	
return { newText: singleLineResult, caretOffset: 3 };	
```

Reply:
ah whoops.